src/main/flight/pid.c

```diff
@@ -869,9 +877,58 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
     rpmFilterUpdate();
 #endif
 
+#ifdef USE_CHIRP
+
+    static int chirpAxis = 0;
+    static bool shouldChirpAxisToggle = true;
+
+    float chirp = 0.0f;
+    float sinarg = 0.0f;
+    if (FLIGHT_MODE(CHIRP_MODE)) {
+
+        // toggle chirp signal logic
+        if (shouldChirpAxisToggle) {
+            shouldChirpAxisToggle = false;
+        }
+
+        // update chirp signal
+        if (chirpUpdate(&pidRuntime.chirp)) {
+            chirp = pidRuntime.chirp.exc;
+            sinarg = pidRuntime.chirp.sinarg;
+        }
+    } else {
+        if (!shouldChirpAxisToggle) {
+
+            // toggle chirp signal logic and increment to next axis for next run
+            shouldChirpAxisToggle = true;
+            chirpAxis++;
+            if (chirpAxis > FD_YAW) {
+                chirpAxis = 0;
+            }
+
+            // reset chirp signal generator
+            chirpReset(&pidRuntime.chirp);
+        }
+    }
```

Comment:
Maybe this is slightly easier to follow:

```suggestion
    static bool shouldChirpAxisToggle = false;

    float chirp = 0.0f;
    float sinarg = 0.0f;
    if (FLIGHT_MODE(CHIRP_MODE)) {
        shouldChirpAxisToggle = true;  // advance chirp axis on next !CHIRP_MODE
        // update chirp signal
        if (chirpUpdate(&pidRuntime.chirp)) {
            chirp = pidRuntime.chirp.exc;
            sinarg = pidRuntime.chirp.sinarg;
        }
    } else {
        if (shouldChirpAxisToggle) {
            // toggle chirp signal logic and increment to next axis for next run
            shouldChirpAxisToggle = false;
            chirpAxis <= FD_YAW ?  (chirpAxis + 1) : 0;
            // reset chirp signal generator
            chirpReset(&pidRuntime.chirp);
            // TODO: reset filter ?
        }
    }
```